src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/CpsProjectSystemReferenceReader.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.References;
+using Microsoft.VisualStudio.Threading;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.ProjectManagement;
+using NuGet.ProjectModel;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Reference reader implementation for the CPS project system in the integrated development environment (IDE).
+    /// </summary>
+    internal class CpsProjectSystemReferenceReader
```

Comment:
What is the lifetime of this? ConfiguredProject is tied to a configuration, so what happens when active configuration is changed?

Reply:
If its tied to the lifetime of the project, then you are going to need to requery the ConfiguredProject over and over again to handle that.

Reply:
It is tied to the lifetime of the project. 

Is there a way to get configuration changes or detect if a configured project is stale?